pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -574,176 +574,59 @@ func CpuWeightToCpuShares(cpuWeight uint64) uint64 {
+	manager, err := libcontainercgroupmanager.New(libcontainerCgroupConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a libcontainer cgroup manager: %w", err)
 	}
-}
 
-func getCgroupMemoryConfig(cgroupPath string) (*ResourceConfig, error) {
-	memLimitFile := "memory.limit_in_bytes"
-	if libcontainercgroups.IsCgroup2UnifiedMode() {
-		memLimitFile = "memory.max"
-	}
-	memLimit, err := fscommon.GetCgroupParamUint(cgroupPath, memLimitFile)
+	cgroups, err := manager.GetCgroups()
 	if err != nil {
-		return nil, fmt.Errorf("failed to read %s for cgroup %v: %v", memLimitFile, cgroupPath, err)
+		return nil, fmt.Errorf("failed to get libcontainer cgroups information: %w", err)
 	}
-	mLim := int64(memLimit)
-	//TODO(vinaykul,InPlacePodVerticalScaling): Add memory request support
-	return &ResourceConfig{Memory: &mLim}, nil
 
-}
-
-// Get the resource config values applied to the cgroup for specified resource type
-func (m *cgroupManagerImpl) GetCgroupConfig(name CgroupName, resource v1.ResourceName) (*ResourceConfig, error) {
-	cgroupPaths := m.buildCgroupPaths(name)
-	cgroupResourcePath, found := cgroupPaths[string(resource)]
-	if !found {
-		return nil, fmt.Errorf("failed to build %v cgroup fs path for cgroup %v", resource, name)
+	cgroupResources := cgroups.Resources
+	if cgroupResources == nil {
+		return nil, fmt.Errorf("cgroup resources are nil for cgroup %v", name)
 	}
-	switch resource {
-	case v1.ResourceCPU:
-		return getCgroupCpuConfig(cgroupResourcePath)
-	case v1.ResourceMemory:
-		return getCgroupMemoryConfig(cgroupResourcePath)
-	}
-	return nil, fmt.Errorf("unsupported resource %v for cgroup %v", resource, name)
-}
 
-func setCgroupv1CpuConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	var cpuQuotaStr, cpuPeriodStr, cpuSharesStr string
-	if resourceConfig.CPUQuota != nil {
-		cpuQuotaStr = strconv.FormatInt(*resourceConfig.CPUQuota, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.cfs_quota_us"), []byte(cpuQuotaStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuQuotaStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUPeriod != nil {
-		cpuPeriodStr = strconv.FormatUint(*resourceConfig.CPUPeriod, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.cfs_period_us"), []byte(cpuPeriodStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuPeriodStr, cgroupPath, err)
-		}
-	}
-	if resourceConfig.CPUShares != nil {
-		cpuSharesStr = strconv.FormatUint(*resourceConfig.CPUShares, 10)
-		if err := os.WriteFile(filepath.Join(cgroupPath, "cpu.shares"), []byte(cpuSharesStr), 0700); err != nil {
-			return fmt.Errorf("failed to write %v to %v: %v", cpuSharesStr, cgroupPath, err)
-		}
-	}
-	return nil
+	return libcontainerCgroupResourcesToResourceConfig(*cgroupResources), nil
 }
 
-func setCgroupv2CpuConfig(cgroupPath string, resourceConfig *ResourceConfig) error {
-	if resourceConfig.CPUQuota != nil {
```

Comment:
This logic doesn't quite match the libcontainer cgroupv2 logic: https://github.com/opencontainers/runc/blob/main/libcontainer/cgroups/fs2/cpu.go#L55-L75